### PR TITLE
Avoid NPE during query parsing

### PR DESCRIPTION
**Pull request by masaruh**
Workaround until Jackson is upgraded to 2.6.x.

### CODE REVIEW

**Comment by masaruh** [comment]
In case #10124 isn't feasible...

**Comment by s1monw** [comment]
I think this looks good! Can you please sign the CLA for this contribution? @kimchy can you please take another look at this too?

File: src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java [hunk 1]
```diff
@@ -88,7 +89,15 @@ public String text() throws IOException {
 
     @Override
     public BytesRef utf8Bytes() throws IOException {
-        return new BytesRef(CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), parser.getTextLength()));
+        // Tentative workaround for https://github.com/elastic/elasticsearch/issues/8629
+        // TODO: Remove this when we upgrade jackson to 2.6.x.
+        char[] chars = parser.getTextCharacters();
+        if (chars == null) {
+            chars = EMPTY_CHARS;
```
Comment (kimchy) [inline src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java]: can we just `return new BytesRef()` in this case? I don't know if the text characters are null if we can really rely on text offset and length? Maybe a better check, not relying on null check is:

```
if (parser.getTextLength() == 0) {
    return new BytesRef();
}
```

**Comment by kimchy** [comment]
I made a small comment here, I think it would be cleaner not to rely on null value for the char array, but use the length and check if its 0. If agreed, +1 on pushing it, thanks @masaruh!

**Comment by masaruh** [comment]
Thanks @kimchy! Addressed your comment along with a few changes to make tests pass (rebase, include license header, extend ElasticsearchTestCase).

File: src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java [hunk 2]
```diff
@@ -88,7 +87,15 @@ public String text() throws IOException {
 
     @Override
     public BytesRef utf8Bytes() throws IOException {
-        return new BytesRef(CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), parser.getTextLength()));
+        // Tentative workaround for https://github.com/elastic/elasticsearch/issues/8629
+        // TODO: Remove this when we upgrade jackson to 2.6.x.
+        char[] chars = parser.getTextCharacters();
```
Comment (kimchy) [inline src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java]: we don't need this now, and can call `parser.getTextCharacters()` in line 96?

**Comment by kimchy** [comment]
@masaruh left another really small comment, other than that, LGTM

File: src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java (hunk 2 above)
Comment (masaruh) [inline src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java]: Very true!

**Comment by masaruh** [comment]
Thanks @kimchy. Fixed and merged.

**Pull request closed without merging**